curator-recipes/src/main/java/org/apache/curator/framework/recipes/queue/ChildrenCache.java

```diff
@@ -66,6 +68,19 @@ public void processResult(CuratorFramework client, CuratorEvent event) throws Ex
         }
     };
 
+    private final ConnectionStateListener connectionStateListener = (__, newState) -> {
+        if ((newState == ConnectionState.CONNECTED) || (newState == ConnectionState.RECONNECTED)) {
+            try
+            {
+                sync();
+            }
+            catch ( Exception e )
+            {
+                throw new RuntimeException(e);
```

Comment:
will this error be logged somewhere ?
as we cannot perform any recovery
will it make more sense to simply log the exception and do not rethrow it ?

Reply:
Yeah, it logs. It's not very obvious but these are handled by `MappingListenerManager` (see https://github.com/apache/curator/blob/master/curator-framework/src/main/java/org/apache/curator/framework/listen/MappingListenerManager.java#L85)